### ISSUE TITLE
not getting accurate answers when i trained with my data

### ISSUE THREAD

**Issue by kkpriyankacoding** [bshao001/ChatLearner#58]

**Comment by kishore0905** [comment]
yes, i am also facing the same problem

**Comment by kishore0905** [comment]
increased the epochs and reduced the learning rate and the mean loss is 0.3 but still can't able to get that accurate result

**Comment by kkpriyankacoding** [comment]
now after reducing my learning rate i was happy that epochs were built but now when i test the data i was unable to get the accurate answer which i predefined, if i ask the same question repeatedly then it is providing the answer at some time.

**Comment by MartinAbilev** [comment]
im not familiar with this repo example of chatbot ... but as far i understand this example ...  it actualy not 100% predefined ....   there is training data  where it answers  sequence containing reference to some function to execute like `what year it is :   its a  _funtion_data_year_ . ' 

maybe in your case there no trigered this type of predefined-answer .     if i get it corectly gradient samples > 0 produce some randomnes in answers making bot behave more "naturaly"

in setings there is numbers of times each corpus need to train on ... maybe ya need to train one where is more `predefined training data`

**Comment by bshao001** [comment]
@MartinAbilev Thank you for helping me to answer the questions.
@kkpriyankacoding @kishore0905 You cannot expect all responses for your given questions (assuming they are exactly same, including punctuation) are the same or even very close to the answers you defined in the training data, unless you manage to train the model until perplexity to 1 (loss is 0). In case your the questions in your training data are not distinct, you still cannot reach that effect. What you can and should expect is to make your model remember responses to those important / critical answers. Keep that in mind, you should now understand why I was keep the data in different folders and make them either repeat or not repeat, or repeat different times during the training.
For further technical questions, please show your support by starring or forking the project. Thank you.

**Comment by kishore0905** [comment]
here its completely matching the whole sentence with question marks also then only it gives the correct answer.what should i do to change the matching process(or how to integrate it with pos tag).

**Comment by bshao001** [comment]
@kishore0905 I am not clear about your question. Did you talk about POS tagging? Please give more details so that I can give more advise.

**Comment by kishore0905** [comment]
Yes I like to add the pos tagging whenever whenever we enter a question
from that we can able to extract the verbs and important nouns based upon
that it will check with the verb and produce the output.here there is no
need of matching with whole sentence. If it is done like that means what
ever the user and even different ways it only extracting the words that
matters the most and matches with it and give the result. This one is my
idea.But I don't know where and how to start.could you help me

On 14-Aug-2018 6:22 PM, "Bo Shao" <notifications@github.com> wrote:

@kishore0905 <https://github.com/kishore0905> I am not clear about your
question. Did you talk about POS tagging? Please give more details so that
I can give more advise.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/bshao001/ChatLearner/issues/58#issuecomment-412860999>,
or mute the thread

**Comment by bshao001** [comment]
This project is not designed to perform POS tagging. For that purpose, you should check on the related publications and look for suitable approaches. In case you have a good reason to develop a chatbot, which requires POS information, and you also prefer to train them jointly (you have an option to use a different module to train for POS), you can as long as you label your training data with the POS information as part of the responses, and you manage to hide the information not for output. As this is far beyond the scope of the project, the details won't be available. You can reach me with wechat or email for further consultation, if you are interested.
